document/FileFormat.mdk

```diff
@@ -2696,31 +2703,318 @@ The tool might provide a "related location" (&sect;[#result-relatedLocations]) w
+|                  |                             | does not return a value.    |                             |
+|------------------|-----------------------------|-----------------------------|-----------------------------|
+| `functionReturn` | Recommended: The            | Optional: If the function   | Optional: If the function   |
+|                  | fully qualified name of     | returns a value or          | returns a value or          |
+|                  | the function being          | values, in which case,      | values, in which case,      |
+|                  | exited.                     | string representations      | the names and values        |
+|                  |                             | of the value or values      | of any variables            | 
+|                  |                             | returned.                   | participating in the        |
+|                  |                             | Absent: If the function     | return expression           |
+|                  |                             | does not return a value.    | or expressions.             |
+|                  |                             |                             | Absent: If the function     |
+|                  |                             |                             | does not return a value.    |
+|------------------|-----------------------------|-----------------------------|-----------------------------|
+| `usage`          | Recommended: The            | Optional: A string          | Absent                      |
+|                  | name of the variable        | representation of the       |                             |
+|------------------|-----------------------------|-----------------------------|-----------------------------|
+
+~
+
+~IsoNumberedExample { @IsoNumberedExample=1 }
+In C++, if the source code contains the declaration
+
+```
+std::string &str = name;
+```
+
+then the value of `kind` would be `"alias"`, the value of `target` would be `"str"`, the value of `parameters` would be
+
+```
+[ "name" ]
+```
+
+and the value of `values` might be
+
+```
+{ "name": "\"John\"" }
+```
+~
+
+~IsoNumberedExample
+In C++, if the source code contains the declaration
+
+```
+std::string &str = name, &str2 = address;
+```
+
+and if the tool creating the log wished to represent both aliases in the log file, then the tool would create
+two `annotatedCodeLocation` objects, each with `kind` set to `"alias"`, and referring to the same source line.
+~
+
+~IsoNumberedExample
+In C++ or C#, if the source code contains the assignment
+
+```
+m = n + p;
+```
+
+then the value of `kind` would be `"assignment"`, the value of `target` would be `"m"`, the value of `parameters` might be
+
+```
+[ "5" ]
+```
+
+and the value of `variables` might be
+
+```
+{  "n": "2", "p": "3" }
+```
+~
+
+~IsoNumberedExample
+In C#, if the source code contains the test
+
+```
+if (s.Length > 0 && y > 2 && valid())
```

Comment:
ok, let's discuss the following case:

if (foo)

someone might think the branch target could refer to 'foo', with parameter of 'false' to express the assumed value. the spec (rightly) forbids this: the target of a branch is the destination, if available. is it worth warning against this mistake? maybe not.